libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -215,6 +215,7 @@ void IRGeneratorForStatements::generate(Block const& _block)
 {
 	try
 	{
+		m_lastLocation = {};
```

Comment:
Why do you reset it here?

Reply:
We are starting a new fresh generation process here and you were concerned earlier whether it needs to be reset or not. 
The line isn't strictly necessary,

Reply:
Ah ok then it's fine!

Reply:
But actually, there are more entry points than just this. I would rely on the constructor to do the job.

Reply:
So please remove this - we should rely on the constructor instead. Having it here but not in the other entry points is confusing.